homeassistant/components/flume/sensor.py

```diff
@@ -137,26 +150,52 @@ def available(self):
 
     @property
     def unique_id(self):
-        """Device unique ID."""
-        return self._device_id
+        """Flume query and Device unique ID."""
+        return f"{self._flume_query_sensor[0]}_{self._device_id}"
 
-    @Throttle(MIN_TIME_BETWEEN_UPDATES)
     def update(self):
         """Get the latest data and updates the states."""
         _LOGGER.debug("Updating flume sensor: %s", self._name)
-        try:
-            self._flume_device.update_force()
-        except Exception as ex:  # pylint: disable=broad-except
-            if self._available:
-                _LOGGER.error("Update of flume sensor %s failed: %s", self._name, ex)
-            self._available = False
-            return
+
+        def format_state_value(value):
+            return round(value, 1) if isinstance(value, Number) else None
+
+        self._flume_data.update()
+        self._state = format_state_value(
+            self._flume_data.flume_device.values[self._flume_query_sensor[0]]
+        )
         _LOGGER.debug("Successful update of flume sensor: %s", self._name)
-        self._state = self._flume_device.value
-        self._available = True
+
+        self._available = self._flume_data.available
 
     async def async_added_to_hass(self):
         """Request an update when added."""
         # We do ask for an update with async_add_entities()
         # because it will update disabled entities
         self.async_schedule_update_ha_state()
+
+
+class FlumeSensorData:
+    """Get the latest data and update the states."""
+
+    def __init__(self, flume_device):
+        """Initialize the data object."""
+        self.flume_device = flume_device
+        self.available = True
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Get the latest data from the Flume."""
+        _LOGGER.debug("Updating Flume data")
+
+        try:
+            self.flume_device.update_force()
+            self.available = True
+            _LOGGER.debug("Returned Flume query: %s", self.flume_device.query_payload)
+            _LOGGER.debug("Returned Flume values: %s", self.flume_device.values)
```

Comment:
Please combine the two debug log calls into one.

Also good if we can move setting available and debug log down below the try... except block.

Reply:
`query_paylod` and `values` is a fairly verbose. I have combined them into 1 call and the debug output now looks like: 

    2020-07-27 20:19:38 DEBUG (SyncWorker_0) [homeassistant.components.flume.sensor] Flume update details: {'values': {'current_interval': 0, 'today': 49.65523134, 'week_to_date': 49.65523134, 'month_to_date': 49.65523134, 'last_60_min': 32.42405452, 'last_24_hrs': 49.65523134, 'last_30_days': 49.65523134}, 'query_payload': {'queries': [{'request_id': 'current_interval', 'bucket': 'MIN', 'since_datetime': '2020-07-27 20:18:00', 'until_datetime': '2020-07-27 20:19:00', 'operation': 'SUM', 'units': 'GALLONS'}, {'request_id': 'today', 'bucket': 'DAY', 'since_datetime': '2020-07-27 00:00:00', 'until_datetime': '2020-07-27 20:19:00', 'operation': 'SUM', 'units': 'GALLONS'}, {'request_id': 'week_to_date', 'bucket': 'DAY', 'since_datetime': '2020-07-27 00:00:00', 'until_datetime': '2020-07-27 20:19:00', 'operation': 'SUM', 'units': 'GALLONS'}, {'request_id': 'month_to_date', 'bucket': 'MON', 'since_datetime': '2020-07-01 00:00:00', 'until_datetime': '2020-07-27 20:19:00', 'units': 'GALLONS'}, {'request_id': 'last_60_min', 'bucket': 'MIN', 'since_datetime': '2020-07-27 19:19:00', 'until_datetime': '2020-07-27 20:19:00', 'operation': 'SUM', 'units': 'GALLONS'}, {'request_id': 'last_24_hrs', 'bucket': 'HR', 'since_datetime': '2020-07-26 20:19:00', 'until_datetime': '2020-07-27 20:19:00', 'operation': 'SUM', 'units': 'GALLONS'}, {'request_id': 'last_30_days', 'bucket': 'DAY', 'since_datetime': '2020-06-27 20:19:00', 'until_datetime': '2020-07-27 20:19:00', 'operation': 'SUM', 'units': 'GALLONS'}]}}